code/modules/mob/living/carbon/carbon_update_icons.dm

```diff
@@ -302,6 +358,29 @@
 
 	return .
 
+/**
+ * Generates a cache key for masks (mainly only used for right legs now, but perhaps in the future...).
+ *
+ * This is exactly like generate_icon_key(), except that it doesn't add `"-[draw_color]"`
+ * to the returned list under any circumstance. Why? Because it (generate_icon_key()) is
+ * a proc that gets called a ton and I don't want this to affect its performance.
+ *
+ * Returns a list of strings.
+ */
+/obj/item/bodypart/proc/generate_mask_key()
+	RETURN_TYPE(/list)
+	. = list()
+	if(is_dimorphic)
+		. += "[limb_gender]-"
+	. += "[limb_id]"
+	. += "-[body_zone]"
+	for(var/obj/item/organ/external/external_organ as anything in external_organs)
+		if(!external_organ.can_draw_on_bodypart(owner))
+			continue
+		. += "-[external_organ.generate_icon_cache()]"
```

Comment:
You could drop the -s here and just .Join("-") to achieve the same result with less headache

Reply:
I'm not quite sure what you mean here.

I'd have to change it for all of those procs, no?

Reply:
see below